File: website/src/models/project.ts
import { TeamMember } from "./team";
import { Collaborator } from "./collaborator";

export interface Project {
    id: number;
    title: string;
    intro: string;
    body: string;
    start?: string;
    end?: string;
    thumbnail?: string;
    osmose_member_contacts?: Array<TeamMember>;
    other_contacts?: Array<string>;
    collaborators?: Array<Collaborator>;
}

File: backend/osmosewebsite/admin/project.py
"""OSmOSE Website - Projects"""
from django.contrib import admin
from django.contrib.admin import ModelAdmin

from backend.osmosewebsite.models import Project


@admin.register(Project)
class ProjectAdmin(ModelAdmin):
    """Projects presentation in DjangoAdmin"""

    list_display = ("title", "intro", "start", "end", "metadatax_project")
    search_fields = ["title"]
    fields = [
        "title",
        "start",
        "end",
        "intro",
        "body",
        "osmose_member_contacts",
        "other_contacts",
        "collaborators",
        "thumbnail",
        "metadatax_project",
    ]
